### PR TITLE
support training, fix some inplace func of nn

### PR DESCRIPTION
**Pull request by Gy-Lu**
This PR should be merged after the other two.

1. Add training code and its shell script, support DDP now.
2. Add lr_scheduler and move loss to hub.
3. Modify some bugs introduced by openfold, the inplace func in `dropout` and the unused `out_product_mean` in `evoformer`.

For unit test, it would cause OOM on CI machine. I am trying to develop a tiny model.

### CODE REVIEW

File: fastfold/model/hub/lr_scheduler.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import torch
```
Comment (Shenggan) [inline fastfold/model/hub/lr_scheduler.py:1]: Add Copyright, with AlQuraishi Laboratory and HPC-Ai

**Comment by Shenggan** [comment]
Seem still have some bugs:

1. `from fastfold.model.loss import *` -> `from fastfold.model.hub.loss import *` in `fastfold/model/nn/heads.py`
2. https://github.com/hpcaitech/FastFold/blob/164f67773f971b57a213d0f402e58e6666fb8a68/fastfold/model/fastnn/template.py#L394 introduced by https://github.com/hpcaitech/FastFold/pull/117

**Review by Shenggan — APPROVED** [review]
